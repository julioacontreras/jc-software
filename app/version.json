{"version":"1696849459988"}