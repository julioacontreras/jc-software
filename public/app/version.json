{"version":"1696848253298"}