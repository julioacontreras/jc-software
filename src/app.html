<!DOCTYPE html>
<html lang="en">
	<head>
		<meta charset="utf-8" />
		<meta name="viewport" content="width=device-width, initial-scale=1" />
		%sveltekit.head%
		<link
			rel="apple-touch-icon"
			sizes="76x76"
			href="%sveltekit.assets%/assets/favicon/apple-touch-icon.png"
		/>
		<link
			rel="icon"
			type="image/png"
			sizes="32x32"
			href="%sveltekit.assets%/assets/favicon/favicon-32x32.png"
		/>
		<link
			rel="icon"
			type="image/png"
			sizes="16x16"
			href="%sveltekit.assets%/assets/favicon/favicon-16x16.png"
		/>
		<link rel="manifest" href="%sveltekit.assets%/assets/favicon/site.webmanifest" />
		<link
			rel="mask-icon"
			href="%sveltekit.assets%/assets/favicon/safari-pinned-tab.svg"
			color="#5bbad5"
		/>
		<meta name="msapplication-TileColor" content="#da532c" />
		<meta name="theme-color" content="#ffffff" />
	</head>
	<body class="theme" data-sveltekit-preload-data="hover">
		<div style="display: contents">%sveltekit.body%</div>
	</body>
</html>
