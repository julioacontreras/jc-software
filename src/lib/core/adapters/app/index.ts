export let app: GlobalAdapter;

export type CallbackGlobal = (data: unknown) => void;

export interface GlobalAdapter {
	subscribe: (callback: CallbackGlobal) => void;

	set: (settings: Record<string, unknown>) => void;
}

export function setApp(instance: GlobalAdapter) {
	app = instance;
}
