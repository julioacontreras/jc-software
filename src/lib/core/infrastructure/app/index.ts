import { writable } from 'svelte/store';
import { setApp, type GlobalAdapter, type CallbackGlobal } from '../../adapters/app';

const settings: Record<string, unknown> = {
	lang: '',
	theme: '',
	activePage: ''
};
const storeGlobal = writable(settings);

class Global implements GlobalAdapter {
	subscribe(callback: CallbackGlobal) {
		storeGlobal.subscribe(callback);
	}

	public set(settings: Record<string, unknown>) {
		storeGlobal.set(settings);
	}
}

setApp(new Global());
