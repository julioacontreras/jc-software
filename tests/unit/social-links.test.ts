import { cleanup, render, screen } from '@testing-library/svelte';
import SocialLinks from '../../src/lib/ui/components/social-links/index.svelte';
import { describe, it, expect, beforeEach, afterEach, vi } from 'vitest';

import { setGlobal } from '../../src/lib/core/adapters/app/index';

const subscribeMocked = vi.fn();

setGlobal({
	subscribe: subscribeMocked,
	set: () => {}
});

describe('SocialLinks.svelte', () => {
	beforeEach(() => {
		subscribeMocked.mockClear();
	});

	afterEach(() => cleanup());

	it('mounts', () => {
		const { container } = render(SocialLinks, {});
		expect(container).toBeTruthy();
		expect(container.innerHTML).toMatchSnapshot();
		expect(subscribeMocked).toBeCalledTimes(1);
	});

	it('Check links', async () => {
		render(SocialLinks, {});
		const linkedin = screen.getByTestId('link-linkedin');
		expect(linkedin).toBeTruthy();
		const github = screen.getByTestId('link-github');
		expect(github).toBeTruthy();
		const medium = screen.getByTestId('link-medium');
		expect(medium).toBeTruthy();
		expect(subscribeMocked).toBeCalledTimes(1);
	});
});
