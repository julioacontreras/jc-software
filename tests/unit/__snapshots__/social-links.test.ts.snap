// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SocialLinks.svelte > mounts 1`] = `"<div><a data-testid=\\"link-linkedin\\" href=\\"https://www.linkedin.com/in/julio-contreras-6579b623/\\"><img src=\\"./assets/svg/linkedin.svg\\" width=\\"39\\" alt=\\"Linkedin\\"></a> <a data-testid=\\"link-github\\" href=\\"https://github.com/julioacontreras\\"><img src=\\"./assets/svg/github.svg\\" width=\\"39\\" alt=\\"Github\\"></a> <a data-testid=\\"link-medium\\" href=\\"https://medium.com/@julio.aranis.contreras\\"><img src=\\"./assets/svg/medium.svg\\" width=\\"39\\" alt=\\"Medium\\"></a></div>"`;
